api/v1beta1/types.go

```diff
@@ -617,9 +622,20 @@ type SubnetSpec struct {
 	// +optional
 	NatGateway NatGateway `json:"natGateway,omitempty"`
 
+	// ServiceEndpoints is a slice of Virtual Network service endpoints to enable for the subnets.
+	// +optional
+	ServiceEndpoints ServiceEndpoints `json:"serviceEndpoints,omitempty"`
```

Comment:
Is this something that would make sense to reuse across multiple clusters when creating a "fleet" of clusters via ClusterClass? If so it should go into `SubnetClassSpec`

Reply:
Honestly I'm not sure but I don't think it would hurt. I'll move it.

Reply:
Updated.